polar-core/src/kb.rs

```diff
@@ -1479,11 +1483,167 @@ mod tests {
                 ..
             })
         ));
+
         // Multiple templates can exist for the same name but only one needs to match
         kb.clear_rules();
         kb.add_rule_type(rule!("f", ["x"; instance!(sym!("Orange"))]));
         kb.add_rule_type(rule!("f", ["x"; instance!(sym!("Orange")), value!(1)]));
         kb.add_rule_type(rule!("f", ["x"; instance!(sym!("Fruit"))]));
         kb.add_rule(rule!("f", ["x"; instance!(sym!("Fruit"))]));
+        assert!(kb.validate_rules().is_empty());
+    }
+
+    #[test]
+    fn test_rule_type_validation_errors_for_non_class_specializers() {
+        let mut kb = KnowledgeBase::new();
+
+        kb.register_constant(sym!("String1"), term!("not an external instance"))
+            .unwrap();
+        kb.register_constant(sym!("String2"), term!("also not an external instance"))
+            .unwrap();
```

Comment:
No need to make any changes but you can have a test return an error type and then use the `?` operator which can sometimes shorten all these `unwrap` s in tests a bit (you may know this).

Reply:
Yeah, I use that sometimes. In this case I'm not too bothered b/c the only `unwrap()`s sprinkled into the actual meat of the test (as opposed to the setup up here) are unwrapping `Option` instead of `Result`